test/unfocused-variable.xspec

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<x:description query="x-urn:test:do-nothing" query-at="do-nothing.xqm" schematron="do-nothing.sch"
+	stylesheet="do-nothing.xsl" xmlns:myv="http://example.org/ns/my/variable"
+	xmlns:x="http://www.jenitennison.com/xslt/xspec">
+
+	<!--
+		Test (/x:description|//x:scenario)/x:variable when they are unfocused by @focus on an inner
+		scenario.
+	-->
+
+	<x:variable name="myv:description-variable" select="'VARIABLE'" />
+
+	<x:scenario label="Outer scenario">
+		<x:variable name="myv:outer-scenario-variable" select="'variable'" />
+
+		<x:scenario focus="focus" label="Focused scenario">
+			<x:call function="false" />
+			<x:expect label="Unfocused description-level variable should hold its original value"
+				select="'VARIABLE'" test="$myv:description-variable" />
+			<x:expect label="Unfocused scenario-level variable in an outer scenario should be empty"
+				select="()" test="$myv:outer-scenario-variable" />
```

Comment:
Looking ahead to #575: I think we should at least consider changing this behavior. A use case I have in mind is where you write your XSpec tests first, and then you gradually work on making the scenarios run and pass, as you develop more and more of the SUT. Nested XSpec structures might lead your in-progress work to have focused scenarios that need to inherit things from outside (such as context, variables, or `x:call` without the parameter children). The inherited parts would need to be processed despite being outside the focused scenario. The test-first workflow becomes awkward if you have to restructure your XSpec file instead of merely changing where your `focus` attributes go.

Reply:
Agreed.
And now I tend to feel `(/x:description|//x:scenario)/(x:param|x:variable)` should be either intact or absent. I'm afraid the current behavior "either intact or empty sequence" would produce weird results silently.

Reply:
By "absent" do you mean "the test author shouldn't have a variable/parameter that can't be evaluated without error, even if it's effectively pending" or "the XSpec compiler shouldn't process a variable/parameter that's effectively pending"?

Reply:
The latter. At run time, pending variables should not be declared at all. That will crash this `x:expect` at line 20-21. But I guess the crash would be better than receiving an empty sequence unknowingly.
And maybe, as you pointed out, not-explicitly-pending variables inherited by focused scenarios should not be deemed pending. I hope #1420 makes it easier to experiment it.

Reply:
> The latter. At run time, pending variables should not be declared at all.

Thanks for clarifying. I agree with you.

Reply:
Addressed in #1452.